### DISCUSSION TITLE
Neue Inhalte erstellen und verbessern

### DISCUSSION THREAD

**Discussion by jukey** [TVTower/TVTower#341] [General]
Wo werden Vorschläge für neue Inhalte (Sendungen, Filme, News, Werbung, etc.) gesammelt und wo können sie am besten diskutiert und kritisiert werden?

### Derzeitige Situation
- Sammlung von Inhalten
  - Derzeit werden Vorschläge in diesem Thread im Forum gesammelt. Das Format dort erschwert Diskussionen, da alle Vorschläge in einer langen Liste Beitrag für Beitrag veröffentlicht werden.
  - Ein anderer Ort an dem Inhalte zu finden sind, ist das [user Verzeichnis](https://github.com/TVTower/TVTower/tree/master/res/database/Default/user). Welche Bewandtnis hat es damit?
- Erstellen von Inhalten
  - Vorlagen für Vorschläge sind nur schwer zu finden.
  - Ebenso ist es nicht einfach eine Anleitung für die Kriterien und Genres zu finden oder
  - eine Verteilung des Sendungen über die Genres und Veröffentlichungsjahre. Diese würde beim setzen von Schwerpunkten beim Erstellen neuer Inhalte helfen.

### Ideen für Verbessrungen
- Schaffung eines neuen Repositories zur Vorbereitung von Inhalten
  - Es könnte Statistiken zu den bereits vorhanden Inhalten enthalten, die dabei helfen die richtigen Schwerpunkte für neue Inhalte zu setzen
  -  Neue Inhalte könnten in einzelnen issues diskutiert werden. Das hätte auch den Vorteil, dass man sich am Anfang auf die kreativen Texte beschränken könnte und später die eher technischen Parameter am einzelenen Inhalt entlang diskutieren könnte. Labels könnten dabei helfen, nach Kategorien (Film, Serie, News, ...) und Genres (Horror, Abenteuer, etc.) zu filtern. 
  - Bestehende (und auch neu hinzukommende) Inhalte im Forum-Thread könnten in issues überführt werden
 
Was denkt ihr?

**Comment by nittka** [comment]
Ich glaube nicht, dass man viele Forennutzer davon überzeugen kann, ein Github-Konto aufzumachen, um Beiträge zu leisten oder hier zu diskutieren (und Discord gibt es ja auch noch). Meiner Ansicht nach wäre es wichtig, die relevanten Diskussionsergebnisse im Forum hier jeweils zeitnah in einem Issue festzuhalten (es schlummern ja auch noch TODOs im Code und in der TODO-Datei im Hauptverzeichnis des Projekts).
Die Datenbankdateien im User-Verzeichnis werden schon vom Spiel verwendet. Man kann seine eigenen Datenbankdateien anlegen und mit diesen spielen - und einige Nutzer habe für ihre Dateien eben einen PR gestellt und kamen damit ins "offizielle Spiel".

Ich weiß nicht, ob ein separates Repository nötig wäre. Warum nicht die bestehenden Möglichkeiten nutzen (Fork, Branch, Pull-Request)? Die Dokumentation und das Tooling könnten natürlich wie überall besser sein (ist aber schwierig, wenn alles von einem allein gemacht werden soll - Priorisierung). Das Dokumentationsthema sehe ich aber auch nicht in einem anderen Repo, sondern eher im Wiki dieses Repository. Wenn man die dort bestehenden Seiten aufräumt, aktualisiert und ergänzt...

Bezüglich Statistiken: sowohl direkt aus den XML-Dateien als auch mit dem entstehenden Xtext-Editor (enthält Generator-Infrastruktur) als auch auf Basis des TVTower-Codes ließe sich sicherlich eine csv-Datei mit aggregierten Infos erzeugen. Man braucht "nur" die richtigen Anforderungen und die Resourcen sie umzusetzen. (Etwas erschwert wird das mit den Veröffentlichungsjahren, da es Einträge mit relativer Jahreszahl gibt - das kann man ja aber mit 1985 als Basisjahr erschlagen.)

**Reply by jukey** [reply]
An ein paar Statistiken würde ich mich mal versuchen. xml2csv scheint ja ungleich simpler als anders herum zu sein.

Ansonsten werde ich mir die verschiedenen Quellen für Dokumentationen, TODOs usw. mal genauer anschauen und (für mich) zusammentragen. :)

Was das Tooling für Inhalte betrifft, verstehe ich, dass es kompliziert ist und finde dein Vorgehen, um schnell Feedback zu sammeln gut.

**Comment by GWRon** [comment]
Die relativen Spieljahrangaben dienen einfach der Verfuegbarkeit...so dass bei jedem Spieljahr was "altes neues" dabei ist.


Aber ja Forennutzer sind eher wenige Discord- bzw. GitHub-Nutzer. Ist schwer, die irgendwie unter einen Hut zu bringen. Ein paar wenige Sachen kamen sogar exklusiv per Mail.

**Comment by nittka** [comment]
Etwas Doku zu den Werten gibt es schon im Wiki (https://github.com/TVTower/TVTower/wiki), ein interessanter Ausgangspunkt für die verschiedenen Flags und Konfigurationsoptionen ist die Konstantenklasse 

https://github.com/TVTower/TVTower/blob/master/source/game.gameconstants.bmx

In jedem "Type " finden sich da die Flags und Namen.

**Reply by nittka** [reply]
Im Zusammenang mit dem Editor, versuche ich gerade, etwas mehr Dokumentation für das Datenbankformat auszuarbeiten. Kopfzerbrechen bereitet mir unter anderem die Stelle, an der sie dann final zu finden sein soll; insb. weil das auch die Strukturierung beeinflusst. Möglichkeiten, die ich aktuell sehe
* Wiki dieses Repos
* misc/documentation in diesem Repo
* GitHub-Pages für dieses Repo erstellen (Layouting aufwändiger)
* direkt das Doku-Repos verwenden und ausbauen (https://github.com/TVTower/Documentation)

Meine aktuellen Vorstellungen für den Grundaufbau der Format-Dokumentation lassen sich auf der Spielwiese im Editor-Repo erahnen (https://github.com/TVTower/TVTDatabaseEditor/tree/docTest/doc). Einstiegspunkt ist database.md; news.md hat schon etwas Inhalt und die Struktur, die ich für die anderen Einträge auch verwenden würde.

Für umfangreichere Dokumentation mag ich das GitHub-Wiki nicht. Es ist ganz nett für Ideensammlung, Notizen etc.. Mir fehlt die Möglichkeit, zusammengehörige Inhalte in Unterverzeichnissen zu strukturieren. Dutzende Dateien auf einer Ebene, die nur durch das Inhaltsverzeichnis strukturiert werden, sind mir zuwider. Bei der Entscheidung zwischen misc/documentation und einem eigenen Repo würde ich persönlich letzteres bevorzugen.

Zumindest bei mir geht das Zusammentragen der Inhalte schneller, wenn die formalen Rahmenbedingungen geklärt sind.

**Reply by GWRon** [reply]
news
> Typ 1 kommt nur zweimal in jorgaeff.xml vor - korrekt und überhaupt benötigt?
> Typ 3 kommt gar nicht vor

Ausgewertet sind sie im Code derzeit nicht - aber Typ 1 wird fuer manche News im Spielcode vergeben (Filmankuendigungen / Bombennews). Hatte das dafuer angedacht, dass die Spiellogik solche Nachrichten nicht "randomisiert" auswaehlt, aber die Datenbank (zwecks Lokalisierung) sie bereithaelt.

Type 3 koennte aber wirklich weg - da das ja nun anders umgesetzt wurde (mittels Script, Availability, live... etc.).



WO zu dokumentieren ist - und Wie? Puh, ich mag das Wiki auch nicht. .md-Dateien sind zwar weniger flexibel anpassbar (immer mit "commit") aber naja ... fuers Nachlesen sicher ausreichend.

**Reply by jukey** [reply]
Für mich klingt es am einfachsten, alles was Dokumentation ist auch im Doku Repository ( https://github.com/TVTower/Documentation ) abzulegen.
Die Dokumentation könnte aufgeteilt werden in
- Dokumentation für Spieler:innen
- Doku für Entwickler:innen

Ein solches Repository könnte später um einen Static Page Generator erweitert werden, der aus den Inhalten eine Doku-Webseite erzeugt. Darüber hinaus ist die Hemschwelle etwas zu editieren in einen Doku-Repo sehr niedrig und kann direkt im Webview in Github erfolgen ohne dass das commit log des eigentlichen Spiel-Quellcodes oder des Editor-Quellcodes mit edit-Nachrichten volläuft.

**Reply by nittka** [reply]
Danke für die Rückmeldungen. Einen offiziellen Status hat das Doku-Repo ja noch nicht. Ich hatte es mal angelegt, um die verteilte Diskussion zu den Eigenschaften und deren Entwicklung von Schauspielern etc. zu konsolidieren (steht immer noch aus).
Aber ja, mein Favorit wäre es auch - zumindest bis ein Stand erreicht ist, den man dann übertragen und dann im Hauptrepo weiterpflegen kann.

Wird außer dem Handbuch noch eine separate Doku für Spieler benötigt? Was könnte da rein? Ggf. sollte man aber zwischen "Programmierer" und Lieferanten sonstiger kreativer Inhalte (DB, Musik, Bilder) unterscheiden.

**Reply by nittka** [reply]
Die Nachrichtenbeschreibung habe ich bis auf Beispiele in der Ausführlichkeit, wie ich sie mir aktuell vorstelle (https://github.com/TVTower/Documentation/blob/dbDoc/database/news.md). Aktuell noch auf dem Branch; Gegenlesen erwünscht.

**Comment by nittka** [comment]
Aktueller Stand:
* Datenbankformat ist dokumentiert
* Nutzbare Version des Datenbankeditors ist erstellt und dokumentiert (Installationsanleitung+Feature-Übersicht)
* Englische Version der Editor-Doku steht noch aus

https://github.com/TVTower/Documentation
